writer.go

```diff
@@ -27,13 +28,18 @@ type Configurer func(*writer)
 
 // writer implements io.Writer
 type writer struct {
-	logger logrus.FieldLogger
-	level  logrus.Level
-	fields map[string]interface{}
+	logger                  logrus.FieldLogger
+	level                   logrus.Level
+	fields                  map[string]interface{}
+	trailingNewLineTrimming bool
 }
 
 // Write will write with the logger, level and fields set in the writer
 func (w *writer) Write(bytes []byte) (int, error) {
+	l := len(bytes)
+	if w.trailingNewLineTrimming && l > 0 && bytes[l-1] == '\n' {
```

Comment:
what do you think about using the std lib [strings.TrimSuffix](https://golang.org/pkg/strings/#TrimSuffix) instead?

Reply:
I'd have to allocate a string for these bytes for that.